worker/firewaller/firewaller.go

```diff
@@ -871,17 +877,23 @@ func (fw *Firewaller) flushInstancePorts(machined *machineData, toOpen, toClose
 	if err != nil {
 		return err
 	}
+	fwInstance, ok := instances[0].(instance.InstanceFirewaller)
+	if !ok {
+		logger.Infof("flushInstancePorts called on an instance of type %T which doesn't support firewall.", instances[0])
```

Comment:
Sorry, I'm an idiot. It didn't occur to me that this method would only be called if the relevant firewall mode were configured. So I think returning an error is sensible after all.